R/parse_expr.R

```diff
@@ -451,6 +493,12 @@ parse_expr_usage <- function(expr, src, call) {
       parse_expr_check_call(expr, src, call)
       args <- lapply(expr[-1], parse_expr_usage, src, call)
       expr <- as.call(c(list(fn), args))
+    } else if (fn_str %in% c("function", "while", "repeat", "for")) {
```

Comment:
I first thought may be `next` is worth forbidding - I might be tempted to write that somewhere...
Then googled and wondered what you thought of...

```
} else if (fn_str != make.names(fn_str)) {
```

Reply:
Hmm, I've not seen that before!

So, the idea here was to avoid a confusing message where speculative/optimistic use of `function` would give a confusing error message `Unsupported function 'function'` which is both awkwardly worded and people don't really think of `function` as a function (it's not really but that's the position it occupies in the tree at that point).

So I disallowed it and then wondered what else to put here.  I do need to ban things that are not valid identifiers in C too, and also in *variables*; we can do that more efficiently by looking at the dependencies.

Given you have your hands full today I might pile all that into this PR...